diff -u a/application_controller.rb b/application_controller.rb
--- a/application_controller.rb	2012-03-28 00:02:28.000000000 +0800
+++ b/application_controller.rb	2012-03-28 00:04:07.000000000 +0800
@@ -67,7 +67,9 @@
   # Returns the current user or nil if no user is logged in
   # and starts a session if needed
   def find_current_user
-    if session[:user_id]
+    if request.env["REMOTE_USER"]
+      (User.find_by_login(request.env["REMOTE_USER"]) rescue nil)
+    elsif session[:user_id]
       # existing session
       (User.active.find(session[:user_id]) rescue nil)
     elsif cookies[:autologin] && Setting.autologin?
